caffe2/CMakeLists.txt

```diff
@@ -516,20 +516,25 @@ endif()
 # ==========================================================
 
 
-
 # Compile exposed libraries.
 add_library(caffe2 ${Caffe2_CPU_SRCS})
 
 
-
 option(TORCH_STATIC "Build libtorch.a rather than libtorch.so" OFF)
 
 
-
 # This is required for older versions of CMake, which don't allow
 # specifying add_library() without a list of source files
 set(DUMMY_EMPTY_FILE ${CMAKE_BINARY_DIR}/empty.cpp)
-file(WRITE ${DUMMY_EMPTY_FILE} "")
+
+if (MSVC)
+  set(DUMMY_FILE_CONTENT "__declspec(dllexport) int ignore_this_library_placeholder(){return 0;}")
```

Comment:
> I don't know why the empty stub is needed here. Maybe it was used to keep the backward compatibility?

Relaying an answer from Dima:
> primarily to keep the shared library file name to be torch, not caffe2 to make sure that people builds and extensions don’t break
Alternatively we could just generate only libtorch instead of libcaffe2 but it’d make the PR bigger and also would still have the same issue for C2 users ( but the latter part might be fine)